doc/tutorial/06_vault.md

```diff
@@ -0,0 +1,143 @@
+# Setup pgagroal-vault
+
+This tutorial will show you how to do setup `pgagroal-vault`.
+
+## Preface
+
+This tutorial assumes that you have already an installation of PostgreSQL 12 (or higher) and `pgagroal` and also assumes that the remote management is enabled in `pgagroal`.
+
+In particular, this tutorial refers to the configuration done in [Install pgagroal](https://github.com/pgagroal/pgagroal/blob/master/doc/tutorial/01_install.md).
+
+Moreover refer to [Enable Remote Management](https://github.com/pgagroal/pgagroal/blob/master/doc/tutorial/03_remote_management.md) to enable remote management and add remote admin user.
+
+## `pgagroal` users setup
+
+Assuming that the master key is already generated and an admin is already present for remote management with `admin` username and `admin1234` password.
+
+### Create a user
+
+As the `pgagroal` operating system user, add the `myuser` to the pooler:
+
+```
+pgagroal-admin -f /etc/pgagroal/pgagroal_users.conf -U myuser -P mypassword user add
+```
+
+The `myuser` and `mypassword` should be the original PostgresSQL's user and its corresponding password.
+
+### Create a frontend user
+
+As the `pgagroal` operating system user, add the `myuser` to the pooler:
+
+```
+pgagroal-admin -f /etc/pgagroal/pgagroal_frontend_users.conf -U myuser -P password user add
+```
+
+**Remember the frontend passwrod should be exactly 8 characters long.**
+
+## `pgagroal-vault` setup
+
+In order to run `pgagroal-vault`, you need to configure the vault `pgagroal_vault.conf` configuration file, that will tell the vault how to work, which address to listen, address of management service in `pgagroal` so on, and then `pgagroal_vault_users.conf` that will instrument the vault about the admin username and password of the remote management.
+
+### Create basic configuration
+
+It is now time to create the main `/etc/pgagroal/pgagroal_vault.conf` configration file, with your editor of choice of using `cat` from the command line, create the following content:
+
+```
+cd /etc/pgagroal
+cat > pgagroal_vault.conf
+[pgagroal-vault]
+host = localhost
+port = 2500
+
+log_type = console
+log_level = info
+log_path = 
+
+[main]
+host = localhost
+port = 2347
+user = admin
+```
+
+and press `Ctrl-D` (if running `cat`) to save the file.
+
+Similarly, create the `/etc/pgagroal/pgagroal_vault_users.conf` file;
+
+```
+cd /etc/pgagroal
+cat > pgagroal_vault_users.conf
+admin:admin1234
+```
+
+and press `Ctrl-D` (if using `cat`) to save the file.
+
+See [the documentation about `pgagroal_vault.conf` for more details](https://github.com/agroal/pgagroal/blob/master/doc/VAULT.md).
+
+
+## Start pgagroal-vault
+
+It is now time to start `pgagroal-vault`, so as the `pgagroal` operating system user run:
+
+```
+pgagroal-vault -c /etc/pgagroal/pgagroal_vault.conf -u /etc/pgagroal/pgagroal_vault_users.conf
```

Comment:
We should high-light that `pgagroal_vault_users.conf` should be the same as `pgagroal_admins.conf`